File: project_euler/problem_030/solution.py
def solve():
    pow5 = [i**5 for i in range(10)]
    total = 0
    for n in range(2, 6 * 9**5 + 1):
        s, t = 0, n
        while t:
            s += pow5[t % 10]
            t //= 10
        if s == n:
            total += n
    print(total)

if __name__ == "__main__":
    solve()

File: project_euler/problem_030/solution.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {
    int pow5[10];
    for (int i = 0; i < 10; i++)
        pow5[i] = i * i * i * i * i;

    int total = 0;
    for (int n = 2; n <= 354294; n++) {
        int s = 0, t = n;
        while (t > 0) {
            s += pow5[t % 10];
            t /= 10;
        }
        if (s == n)
            total += n;
    }

    cout << total << endl;
    return 0;
}
